ec-service/src/test/java/org/dcsa/endorsementchain/service/PartyServiceTest.java

```diff
@@ -24,7 +25,8 @@
 
 @ExtendWith(MockitoExtension.class)
 class PartyServiceTest {
-  @Spy private PartyMapper partyMapper = Mappers.getMapper(PartyMapper.class);
+  @Spy private SupportingPartyCodeMapper supportingPartyCodeMapper = Mappers.getMapper(SupportingPartyCodeMapper.class);
+  @InjectMocks @Spy private PartyMapper partyMapper = Mappers.getMapper(PartyMapper.class);
```

Comment:
Don't think you need `@InjectMocks` here, can be removed.

Reply:
If I remove `@InjectMocks` the test fails on a NPE inside the PartyMapper because it uses another mapper that is not injected into it. :-/

Reply:
Interesting, I didn't realize that. I think a way around that is also to use `@Spy` or `@Mock` on the dependency mapper class. 
Depending on the test use case, if you are validating the mapping for the nested object the above might be useful.
If not, and the current implementation satisfies your use case then I will close the comment.